File: 数组/2.统计数组的元素/448.js
/*
 * @Description: 
 * @Author: young
 * @Date: 2022-08-15 17:14:11
 * @LastEditTime: 2022-08-15 18:28:30
 * @FilePath: \young_leetcode\数组\2.统计数组的元素\448.js
 */


// const findDisappearedNumbers = function(nums) {
//     const length = nums.length;
//     const new_arr = [];
//     nums = nums.sort();
    
//     for (let i=0;i<length;i++) {
//         new_arr.push(i+1);
//     }

//     let i = 0;
//     const res = [];

//     while (i<length) {
//         if (new_arr[i] !== nums[i]) {
//             res.push(new_arr[i]);
//             i++;
//         }
//         i++;
//     }

//     return res;
// } 

const findDisappearedNumbers = function(nums) {
    for (const num of nums) {
        nums[num-1] = -Math.abs(nums[Math.abs(num)-1]);
    }

    console.log(nums)

    const res = [];

    for (let i=0;i<nums.length;i++) {
        if (nums[i]>0) {
            res.push(i+1);
        }
    }

    return res;
}

console.log(findDisappearedNumbers([4,3,2,7,8,2,3,1]))

File: 数组/2.统计数组的元素/448.py
'''
Description: 找到所有数组中消失的数字
Author: young
Date: 2022-08-15 16:53:56
LastEditTime: 2022-08-15 17:14:00
FilePath: \young_leetcode\数组\2.统计数组的元素\448.py
'''

"""
    1.根据nums的长度，创建一个新的包含1到n的数组
    2.找出新数组包含但是旧数组不包含的元素，作为数组返回
"""
def findDisappearedNumbers(nums):
    return list(set([i+1 for i in range(len(nums))]).difference(set(nums)))


print(findDisappearedNumbers([4,3,2,7,8,2,3,1]))
print(findDisappearedNumbers([1,1]))